src/services/date-utilities.ts

```diff
@@ -254,4 +254,60 @@ export default class DateUtilities {
     return dateTo.getMonth() - dateFrom.getMonth() +
       (12 * (dateTo.getFullYear() - dateFrom.getFullYear()))
   }
+
+  /**
+   * Add a number of months to a date and return "YYYY-MM-DD".
+   * Date must be in the format of "YYYY-MM-DD" and months is a number
+   * @example (3, 2023-02-03) -> "2023-05-03"
+   * @example (18, 2023-02-03) -> "2024-08-03"
+   */
+  static addMonthsToDate (month: number, date: string): string {
+    if (!date) {
+      date = '2023-01-01'
+    }
+    const temp = this.yyyyMmDdToDate(date)
+    temp.setMonth(temp.getMonth() + month)
+    const dateAfterAddition = this.dateToYyyyMmDd(temp)
+    return dateAfterAddition
+  }
+
+  /**
+   * Subtract a number of months to a date and return "YYYY-MM-DD".
+   * Date must be in the format of "YYYY-MM-DD" and months is a number
+   * @example (3, 2023-05-03) -> "2023-02-03"
+   * @example (18, 2024-08-03) -> "2023-02-03"
+   */
+  static subtractMonthsToDate (month: number, date: string): string {
+    if (!date) {
+      date = '2023-01-01'
+    }
+    const temp = this.yyyyMmDdToDate(date)
+    temp.setMonth(temp.getMonth() - month)
+    const dateAfterSubtraction = this.dateToYyyyMmDd(temp)
+    return dateAfterSubtraction
+  }
+  /**
+   * Decrease one date from another and return number of months as the difference.
+   * Dates must be in the "YYYY-MM-DD" format
+   * @example (2023-02-03, 2024-08-03) -> 18
+   * @example (2023-02-03, 2023-04-03) -> 2
+   */
+  static subtractDates (dateFrom: string, dateTo: string): number {
+    if (!dateFrom) {
+      dateFrom = '2023-01-01'
+    }
+    if (!dateTo) {
+      dateTo = this.addMonthsToDate(24, dateFrom)
+    }
+    const expiryDate = this.yyyyMmDdToDate(dateTo)
+    const currDate = this.yyyyMmDdToDate(dateFrom)
+    const monthDiff = expiryDate.getMonth() - currDate.getMonth()
+    const yearDiff = (12 * (expiryDate.getFullYear() - currDate.getFullYear()))
+    let difference = monthDiff + yearDiff
+    // For example if Jan 31st and March 1, difference is 1 not 2
+    if ((expiryDate.getDate() + 25) < currDate.getDate()) {
```

Comment:
Why add 25?

Reply:
This is like a cut-off value. Since some months have more days than others, the difference between them might be incorrect if I don't add this 25 value.

For example, take a look at this scenario:
Current date: Jan 31st
Expiry Date: March 1st.
If I do `((expiryDate.getDate()) < currDate.getDate())`, it's going to be 1 < 31, we return 1 which is correct.
With the 25:
26 < 31, we return 1 which is correct.
In this case, it doesn't matter. 

Now take a look at this scenario:
Current date: Jan 16th
Expiry Date: March 15th.
If I do `((expiryDate.getDate()) < currDate.getDate())`, it's going to be 15 < 16, we return 1 which is incorrect!
Now with the 25:
40 < 16, we return 2 since we didn't go and decrement the difference. We return in this case the correct value!

This was done in Create UI when I was doing the limited restoration creation. I've tested this extensively. Let me know if that clears things up!

Reply:
Please add a comment explaining why you use 25.